images/bootstrap/runner.sh

```diff
@@ -13,20 +13,22 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -o errexit
-set -o nounset
-set -o pipefail
-
-# get test-infra for latest bootstrap etc
-git clone https://github.com/kubernetes/test-infra
-
+# generic runner script, handles DIND, bazelrc for caching, etc.
 
 # Check if the job has opted-in to bazel remote caching and if so generate 
 # .bazelrc entries pointing to the remote cache
 export BAZEL_REMOTE_CACHE_ENABLED=${BAZEL_REMOTE_CACHE_ENABLED:-false}
 if [[ "${BAZEL_REMOTE_CACHE_ENABLED}" == "true" ]]; then
     echo "Bazel remote cache is enabled, generating .bazelrcs ..."
-    ./test-infra/images/bootstrap/create_bazel_cache_rcs.sh
+    # if we have a test-infra checkout (because bootstrap), use that since
+    # it is newer and this is probably k/k so we can push fixes faster this way
+    # otherwise run the one baked into the image
+    # TODO(bentheelder): someday only support the pod-utils
+    if [[ -d "./test-infra" ]]; then
+        ./test-infra/images/bootstrap/create_bazel_cache_rcs.sh
+    else
+        /usr/local/bin/create_bazel_cache_rcs.sh | bash
```

Comment:
we have to pipe the output into bash? (we don't do this on line 28)

Reply:
oops, yeah, I initially thought about curl | bash the latest from test-infra but felt terrible about that and did this instead. forgot to drop the pipe to bash 🤦‍♂️ 

thanks!

Reply:
fixed